### PR TITLE
wip: ignore dev code artifacts for CSS only updates

### PR DESCRIPTION
**Pull request by rixo**
Filter out dev only code for the CSS only test, enabling more CSS only updates.

We'd still need to make this feature optional (opt out, I'd say), because the implementation is not risk free. There are some rare corner cases where we will collide with user code (resulting in CSS only updates, where a true one was expected).

### CODE REVIEW

File: packages/vite-plugin-svelte/src/handleHotUpdate.ts [hunk 1]
```diff
@@ -62,26 +63,41 @@ export async function handleHotUpdate(
   return result
 }
 
+function normalizeNonCss(code: string, cssHash: string) {
+  // trim HMR transform
+  const indexHmrTransform = code.indexOf(
+    'import * as ___SVELTE_HMR_HOT_API from'
+  )
+  if (indexHmrTransform !== -1) code = code.slice(0, indexHmrTransform)
```
Comment (dominikg) [inline packages/vite-plugin-svelte/src/handleHotUpdate.ts:71]: this assumes the indexHmrTransform is always at the start, tying this logic to impl in svelte-hmr

File: packages/vite-plugin-svelte/src/handleHotUpdate.ts [hunk 2]
```diff
@@ -62,26 +63,41 @@ export async function handleHotUpdate(
...
+  // remove irrelevant bits
+  // NOTE the closer we're making our regexes to what we know of the compiler's
+  //      output, the more fragile we are, but the less we risk catching
+  //      similarly looking user's code
+  return (
+    code
+      // ignore css hashes in the code (that have changed, necessarily)
+      .replace(new RegExp('\\s*\\b' + cssHash + '\\b\\s*', 'g'), '')
+      // TODO this one might need to be a little more specific
+      .replace(/\s*attr_dev\([^,]+,\s*"class",\s*""\);?\s*/g, '')
```
Comment (dominikg) [inline packages/vite-plugin-svelte/src/handleHotUpdate.ts:81]: whats this about? is it changing if you add a new class to css?

File: packages/vite-plugin-svelte/src/handleHotUpdate.ts (hunk 1 above)
Comment (rixo) [reply packages/vite-plugin-svelte/src/handleHotUpdate.ts:70]: This assume the transform is at the end, but yeah it ties it horribly to an implementation detail in svelte-hmr. That's why I believe this function should live in svelte-hmr. I'm in the process of doing that and PR an update.

**Comment by dominikg** [comment]
i've added the changes with some additional logging to main.
